frontend/src/lib/components/SelectGradientOverflow.tsx

```diff
@@ -118,30 +119,50 @@ export function SelectGradientOverflow({
         }
     }
     document.addEventListener('click', outsideClickListener)
+
     return (
         <div ref={containerRef} style={{ width: '100%' }}>
-            <Select
-                {...props}
-                dropdownAlign={placement ? ANTD_TOOLTIP_PLACEMENTS[placement] : undefined}
-                ref={selectRef}
-                open={isOpen}
-                onFocus={onFocus}
-                onBlur={onBlur}
-                onPopupScroll={() => {
-                    updateScrollGradient()
+            <ConfigProvider
```

Comment:
This feels magical... What does it affect exactly?

Reply:
Yup, I was also able to read the docs and see what this component does :).

My point is that it's not clear from the code, which component of which deeply nested subcomponent benefits from this `renderEmpty`, and why is it better than just passing down the prop directly. Can you clarify that part in the code? Something like:

```
// This ConfigProvider is used to pass prop "renderEmpty" down to a list in the Select > dropdownRender > DropdownGradientRenderer > bla > xyz > List ... > Empty
// and this is a better approach than passing down the prop itself because X, Y, Z
```

... and a corresponding `// here be dragons` on the other end as well. Magically having props be different than what's specified is... magical.

Reply:
I'm very literal :)

Reply:
Comment added. My understanding is that it is being applied to the Select not one of its children

Reply:
Unless I'm being silly it's not on Select https://ant.design/components/select/

But happy to be proved wrong!